cpp/include/rapidsmpf/communicator/single.hpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <rapidsmpf/buffer/buffer.hpp>
+#include <rapidsmpf/buffer/resource.hpp>
+#include <rapidsmpf/communicator/communicator.hpp>
+#include <rapidsmpf/config.hpp>
+
+namespace rapidsmpf {
+
+/**
+ * @brief Single process communicator class that implements the `Communicator` interface.
+ *
+ * This class stubs out the `Communicator` interface with functions that throw.
+ * When sending to/receiving from self the internal logic should move
+ * buffers through the shuffler, rather than invoking send/recv.
+ */
+class Single final : public Communicator {
+  public:
+    /**
+     * @brief Represents the future result of an operation.
+     *
+     * This class is used to handle the result of a communication operation
+     * asynchronously.
+     */
+    class Future : public Communicator::Future {
+        friend class Single;
+
+      public:
+        ~Future() noexcept override = default;
+    };
+
+    /**
+     * @brief Construct a single process communicator.
+     *
+     * @param options Configuration options.
+     */
+    Single(config::Options options);
+
+    ~Single() noexcept override = default;
+
+    /**
+     * @copydoc Communicator::rank
+     */
+    [[nodiscard]] constexpr Rank rank() const override {
+        return 0;
+    }
+
+    /**
+     * @copydoc Communicator::nranks
+     */
+    [[nodiscard]] constexpr Rank nranks() const override {
+        return 1;
+    }
+
+    /**
+     * @copydoc Communicator::send
```

Comment:
Should the exceptions be documented here and other methods, since they don't exist in the copied doc?